job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/SchedulerImpl.java

```diff
@@ -263,10 +259,10 @@ public void start() {
                             return Observable.just(time);
                         });
                     })
-//                    .doOnNext(activeTimeSlices::remove)
                     .subscribe(
                             d -> {
                                 logger.debug("Finished post job execution clean up for [" + d + "]");
+                                // TODO should this be in a synchronized block?
```

Comment:
Not unless you need some return values. Both threads can do remove and ifPresent they can do something else. If you needed to check that if something succeeds, then do something then this would be a different case. Why can't two threads execute both of these?